pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +376,70 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,shootTechnicalID to global targetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() {
+	if len(targetInfo) > 0 {
+		return
+	}
 	var target Target
 	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		targetInfo[string(t.Kind)] = string(t.Name)
+	}
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
+	shoot, err := getShootObject()
 	checkError(err)
-	return shoot.Status.TechnicalID
+	targetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+	// user for case garden-project-shoot add [seed] = seedName
+	if isTargeted("project", "shoot") {
+		targetInfo["seed"] = *shoot.Spec.SeedName
+	}
+}
```

Comment:
How about the following implementation. It tries to set the values to targetInfo, according to what is available on the target stack. It does not guarantee that all garden,project,seed,shoot,shootTechnicalID values are set. This can't be done anyhow and depends on which information is available from the target stack.
```go
/*
getTargetMapInfo tries to set garden,project,seed,shoot,shootTechnicalID to global targetInfo. Use getFromTargetInfo to fetch values.
*/
func getTargetMapInfo() {
	if len(targetInfo) > 0 {
		return
	}
	var target Target
	ReadTarget(pathTarget, &target)
	for _, t := range target.Stack() {
		targetInfo[string(t.Kind)] = string(t.Name)
	}

	if isTargeted("shoot") {
		shoot, err := getShootObject()
		checkError(err)
		targetInfo["shootTechnicalID"] = shoot.Status.TechnicalID

		if targetInfo["seed"] == "" {
			targetInfo["seed"] = *shoot.Spec.SeedName
		}
	}

	if targetInfo["project"] == "" {
		projectObj, err := getProjectObject()
		checkError(err)
		targetInfo["project"] = projectObj.Name
	}
}
```

Reply:
done